tests/test_retention_policies.py

```diff
@@ -180,49 +181,74 @@ def test_backup_status(self, server):
 
         # Build a BackupInfo object with status to DONE
         backup_info = build_test_backup_info(
-            server=server, backup_id="test1", end_time=datetime.now(tzlocal())
+            server=server,
+            backup_id="test_backup",
+            end_time=datetime.now(tzlocal()),
+            parent_backup_id=None,
+            children_backup_ids=["test_backup_child"],
+        )
+
+        # Build a CHILD BackupInfo object with status to DONE
+        child_backup_info = build_test_backup_info(
+            server=server,
+            backup_id="test_backup_child",
+            end_time=datetime.now(tzlocal()) + timedelta(days=1),
+            parent_backup_id="test_backup",
+            children_backup_ids=None,
         )
 
         # instruct the get_available_backups method to return a map with
         # our mock as result and minimum_redundancy = 1
-        server.get_available_backups.return_value = {"test_backup": backup_info}
+        server.get_available_backups.return_value = {
+            "test_backup": backup_info,
+            "test_backup_child": child_backup_info,
+        }
         server.config.minimum_redundancy = 1
-        # execute retention policy report
-        report = rp.backup_status("test_backup")
+        # execute retention policy report on parent
+        report_parent = rp.backup_status("test_backup")
+
+        assert report_parent == "VALID"
+
+        # execute retention policy report on child
+        report_child = rp.backup_status("test_backup_child")
 
-        assert report == "VALID"
+        assert report_child == "VALID"
         # Force context of retention policy for testing purposes.
         # Expect the method to return a BackupInfo.NONE value
         rp.context = "invalid"
-        empty_report = rp.backup_status("test_backup")
+        empty_report_parent = rp.backup_status("test_backup")
+
+        assert empty_report_parent == BackupInfo.NONE
+
+        empty_report_child = rp.backup_status("test_backup_child")
 
-        assert empty_report == BackupInfo.NONE
+        assert empty_report_child == BackupInfo.NONE
 
         rp = RetentionPolicyFactory.create(
             "retention_policy", "RECOVERY WINDOW OF 4 WEEKS", server=server
         )
         assert isinstance(rp, RecoveryWindowRetentionPolicy)
 
-        # Build a BackupInfo object with status to DONE
-        backup_info = build_test_backup_info(
-            server=server, backup_id="test1", end_time=datetime.now(tzlocal())
-        )
+        # execute retention policy report on parent
+        report_parent = rp.backup_status("test_backup")
 
-        # instruct the get_available_backups method to return a map with
-        # our mock as result and minimum_redundancy = 1
-        server.get_available_backups.return_value = {"test_backup": backup_info}
-        server.config.minimum_redundancy = 1
```

Comment:
I can see you refactored this test a bit to avoid duplication of commands when setting up the mocks.

Although I think this like `server.config.minimum_redundancy = 1` should not be removed as part of that refactoring.

Reply:
Never mind, I can see this is also set above, this was a duplication in the original code.